File: Command/Python/TurnOnLight.py
# -*- coding: utf-8 -*-
"""
Created on Wed May 18 20:52:01 2022

@author: Mahmoud Saeed
"""

import Command as command

class TurnOnLight(command.Command):
    
    light = None
    def __init__(self , light):
        self.light = light
        
    def excute(self):
        self.light.turnon()
    
    def undo(self):
        self.light.turnoff()
    
    

File: Command/Java/src/command/turnOnLight.java
package command;

public class turnOnLight implements Command{
    private Light light;
    public turnOnLight(Light light){
        this.light = light;
    }


    @Override
    public void execute() {
        this.light.turnOn();
    }

    @Override
    public void undo() {
        this.light.turnOff();

    }
}
